traffic-advice.bs

```diff
@@ -0,0 +1,265 @@
+
+While [[FETCH]] is used to describe the algorithm to request this resource, such agents might not implement [[HTML]].
+
+Definitions {#dfns}
+===================
+
+A <dfn>traffic advice entry</dfn> is a [=struct=] with the following [=struct/items=]:
+*  <dfn for="traffic advice entry">disallowed flag</dfn>, a [=boolean=] which is initially false
+   <div class="note">If the [=traffic advice entry/disallowed flag=] is true, the advice requests that traffic, including establishing connections and sending requests, be avoided.</div>
+
+A <dfn>traffic advice result</dfn> is null, a [=traffic advice entry=], or `"unreachable"`.
+
+An <dfn>agent identity</dfn> is a [=list=] of [=strings=]. It must contain at least two elements, and the last must be `"*"`.
+
+Identity {#identity}
+====================
+
+Each agent should have an brand name that specifically identifies it (such as `PollyPrefetchProxy`).
+
+Its [=agent identity=] is all of the following that apply, in order:
+
+1.  The brand name
+1.  `"prefetch-proxy"`, if the agent is a proxy server which exclusively serves prefetch traffic (for example, a [private prefetch proxy](https://github.com/buettner/private-prefetch-proxy))
+1.  `"*"`
+
+Fetching {#fetching}
+====================
+
+<section algorithm="generate-url">
+
+To <dfn>generate a traffic advice URL</dfn> for [=origin=] |origin|, run the following steps:
+
+1.  If |origin| is not a [=tuple origin=], return failure.
+
+1.  If |origin|'s [=origin/scheme=] is not an [=HTTP(S) scheme=], return failure.
+
+1. If |origin| is not a [=potentially trustworthy origin=], return failure.
+
+1.  Return a new [=URL=] as follows:
+
+    :  [=url/scheme=]
+    :: |origin|'s [=origin/scheme=]
+    :  [=url/host=]
+    :: |origin|'s [=origin/host=]
+    :  [=url/port=]
+    :: |origin|'s [=origin/port=]
+    :  [=url/path=]
+    :: « `".well-known"`, `"traffic-advice"` »
+    
+</section>
+
+<section algorithm="fetch-traffic-advice">
+
+To <dfn>fetch traffic advice</dfn> for [=origin=] |origin|, [=agent identity=] |identity| and algorithm |whenComplete| accepting a [=traffic advice result=]:
+
+1.  Let |url| be the result of [=generating a traffic advice URL=] for |origin|.
+    If it results in failure, then return failure.
+
+1.  Let |request| be a [=request=] as follows:
+
+    :  [=request/method=]
+    :: `` `GET` ``
+    :  [=request/URL=]
+    :: |url|
+    :  [=request/client=]
+    :: null
+    :  [=request/credentials mode=]
+    :: `"omit"`
+    :  [=request/redirect mode=]
+    :: `"manual"`
+        <div class="note">This means that a [=redirect status=] will not lead to another origin being contacted.</div>
+
+1.  Let |processResponse| be the following steps, given [=response=] |response|:
+
+    1.  If |response|'s [=response/type=] is `"error"`, then terminate the fetch, run |whenComplete| with `"unreachable"`, and return.
```

Comment:
Terminate should link to https://fetch.spec.whatwg.org/#concept-fetch-terminate . (Although I'm not sure it's necessary in this particular step since a network error means the fetch is basically over now.)

Reply:
I had spent some time doing battle to try and get Bikeshed to produce this link. `[=fetch/terminate=]` finally did the trick, though.